src/git-extract

```diff
@@ -59,10 +59,10 @@ function steps {
 
   echo "checkout_main_branch"
   echo "rebase_tracking_branch"
-  echo_if_true "push" "$HAS_REMOTE"
+  echo_if_true "push_branch $MAIN_BRANCH_NAME" "$HAS_REMOTE"
```

Comment:
Tried converting this and the two steps above to `sync_branch_steps` but ran into a very interesting issue which I want to deal with separately.